packages/replay/src/utils.ts

```diff
@@ -88,12 +93,64 @@ export async function exponentialBackoffRetry<T>(
       if (currentAttempt == MAX_ATTEMPTS) {
         throw e;
       }
-      await waitForTime(backoff(currentAttempt));
+      await waitForTime(backOffStrategy(currentAttempt));
     }
   }
   throw Error("ShouldBeUnreachable");
 }
 
+export async function exponentialBackoffRetry<T>(
+  fn: () => Promise<T>,
+  onFail?: (e: unknown) => void
+): Promise<T> {
+  return retry(fn, geometricBackoff, onFail);
+}
+
+export async function linearBackoffRetry<T>(
```

Comment:
Maybe add a unit test for this? Most of the important logic has already been handled in existing tests